### PR TITLE
ffi_unwind_calls: treat RustIntrinsic like regular Rust calls

### PR DESCRIPTION
**Pull request by RalfJung**
Also add some comments to `abi_can_unwind` to explain what happens.

r? @nbdd0121  Cc @BatmanAoD

### CODE REVIEW

**Comment by RalfJung** [comment]
Ah, sorry, I thought someone must have made you reviewer by now *oops*.
r? compiler

File: compiler/rustc_mir_transform/src/ffi_unwind_calls.rs [hunk 1]
```diff
@@ -10,6 +10,10 @@ use rustc_target::spec::PanicStrategy;
 
 use crate::errors;
 
+/// Some of the functions declard as "may unwind" by `fn_can_unwind` can't actually unwind. In
+/// particular, `extern "C"` is still considered as can-unwind on stable, but we need to to consider
+/// it cannot-unwind here. So below we check `fn_can_unwind() && abi_can_unwind()` before cincluding
```
Comment (nbdd0121) [inline compiler/rustc_mir_transform/src/ffi_unwind_calls.rs:15]:
```suggestion
/// it cannot-unwind here. So below we check `fn_can_unwind() && abi_can_unwind()` before concluding
```

**Review by nbdd0121 — APPROVED** [review]

**Comment by BatmanAoD** [comment]
> Ah, sorry, I thought someone must have made you reviewer by now _oops_. 

I don't actually know very much at all about rustc internals, and haven't read much of the code.

File: compiler/rustc_mir_transform/src/ffi_unwind_calls.rs [hunk 2]
```diff
@@ -10,6 +10,10 @@ use rustc_target::spec::PanicStrategy;
 
 use crate::errors;
 
+/// Some of the functions declard as "may unwind" by `fn_can_unwind` can't actually unwind. In
```
Comment (davidtwco) [inline compiler/rustc_mir_transform/src/ffi_unwind_calls.rs:13]:
```suggestion
/// Some of the functions declared as "may unwind" by `fn_can_unwind` can't actually unwind. In
```

**Review by davidtwco — APPROVED** [review]
r=me with typo fix

**Pull request merged**